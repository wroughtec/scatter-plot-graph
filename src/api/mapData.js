const newLine = /\r\n|\r|\n/;

const mapData = data => {
  const coordinates = {};

  if (data) {
    data
      .trim()
      .split(newLine)
      .map((item, i) => {
        const columns = item.split(',');

        if (i === 0) {
          return columns.forEach(column => {
            coordinates[column] = [];
          });
        }

        return Object.keys(coordinates).forEach((key, index) => {
          coordinates[key].push(parseFloat(columns[index]));
        });
      });
  }

  return coordinates;
};

module.exports = mapData;
