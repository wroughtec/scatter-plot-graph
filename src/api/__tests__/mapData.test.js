import mapData from '../mapData';

describe('mapData', () => {
  const data = `
    x,y
    18.71230688,11034.12
    13.61320583,16340.26
    17.58212046,25200.85
    0.33767329,503.22
    11.34590321,7029.99
    65.25484819,38744.4
  `;

  it('should convert map to JSON', () => {
    expect(mapData(data)).toMatchSnapshot();
  });

  it('should return empty object when no data', () => {
    expect(mapData()).toEqual({});
  });
});
