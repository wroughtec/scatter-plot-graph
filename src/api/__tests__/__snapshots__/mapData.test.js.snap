// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapData should convert map to JSON 1`] = `
Object {
  "x": Array [
    18.71230688,
    13.61320583,
    17.58212046,
    0.33767329,
    11.34590321,
    65.25484819,
  ],
  "y": Array [
    11034.12,
    16340.26,
    25200.85,
    503.22,
    7029.99,
    38744.4,
  ],
}
`;
